SO_properties.py

```diff
@@ -1313,6 +1584,8 @@ def calculate(self, input_halo, search_radius, data, halo_result):
             name = prop[0]
             outputname = prop[1]
             description = prop[5]
+            if ("CoreExcised" in name) & (self.SO_name != "500_crit"):
```

Comment:
Is that very robust and future-proof? Seems like looking for a name pattern is a bit dangerous. 

Also, what happens if one wants to change the SO used? I guess the code needs to be changed in a few places?

Reply:
This is not super future proof indeed, but we asked Bert about the best way to do this and this was the solution we thought was the best. There should be a better way to do this in the Colibre updates to SOAP, but we decided to use this "hack" to get it done at a reasonable time-frame (also as this version is stable for FLAMINGO).

Reply:
I was going to suggest adding a parameter to the SOProperties class: something like excise_core_fraction=None by default and then we can set it to 0.15 when we create the instance for 500crit in compute_halo_properties.py. And we could only leave the core excised quantities out of the property list if excise_core_fraction is None.

Reply:
What Bert told us is that the colibre version of SOAP has a parameter file which allows you to selectively turn on properties. At the moment that is only supported for all apertures of a halo type, but Bert thinks it shouldn't be to hard to change this and make it such that you can selectively turn on properties for specific apertures only from the parameter file. However, that is not currently available yet, hence our choice for this faster method

Reply:
Something like this: https://github.com/SWIFTSIM/SOAP/tree/core_excision_fraction

Reply:
I think a more complete solution might be nicer so I'm happy to go with that solution. We made it simple so it's easy to replace once the colibre SOAP version get's merged

Reply:
I can see why Bert suggested doing this in a simple way now. Adding a parameter to the SOProperties class which changes the property list is a problem because the documentation script tries to extract the property list without creating a class instance, so the property list can't depend on `__init__` parameters.

I can make it work by having a CoreExcisedSOProperties sub-class which is just an SOProperties with extra entries in the property list. I'll try to test it more tomorrow.

Reply:
I've pushed the changes which make the core excision fraction a parameter. I also fixed the duplicate dataset name issue below, and with that fix my modifications don't change the output.

Reply:
Great, is there a SOAP catalog with all the changes somewhere on Cosma I can have a look at to check all the fields?

Reply:
PartType0/ComptonYParameters in the snapshot does have zeros in it.

Reply:
Hmm that's odd, does it also give those errors for the core-excised versions of those quantities? I Don't think there is an AGN filter for the snapshots.

Reply:
No, I think I've only seen lines 690 and 700 (Tgas_cy_weighted and Tgas_cy_weighted_no_agn) mentioned in the logs. Although the run hasn't finished yet.

Reply:
So only particles that are starforming are set to zero for compton y. So potentially this happens for some weird galaxies in the smallest apertures where all the gas is starforming. Maybe we should add a catch where Ngas == Nsf?

Reply:
Or more precisely, return none if Ngas <= N_gas_SFR

Reply:
How about just computing the denominator first and if it's zero we return None?

Reply:
We have another divide by zero in the SpectroscopicLikeTemperature functions because the electron number density can be zero. I can fix that in the same way.

Reply:
It runs without raising any exceptions now. There's output in
```
/snap8/scratch/dp004/jch/FLAMINGO/ScienceRuns/L1000N0900/HYDRO_FIDUCIAL/halo_properties
/snap8/scratch/dp004/jch/FLAMINGO/ScienceRuns/L1000N0900/DMO_FIDUCIAL/halo_properties
```

Reply:
Could we also process a `m9` resolution run? Might be more suitable for Scott and others to do tests on and verify it's all OK.

Reply:
I've submitted jobs to do a couple of snapshots from L1000N1800/HYDRO_FIDUCIAL.

Reply:
There are now outputs for z=0 and z=1 of L1000N1800/HYDRO_FIDUCIAL in `/snap8/scratch/dp004/jch/FLAMINGO/ScienceRuns/L1000N1800/HYDRO_FIDUCIAL/halo_properties`.

Reply:
Are we happy that the output from this is ok now? If so I think it's ready to merge.